ciao-controller/openstack_volume.go

```diff
@@ -288,11 +280,19 @@ func (c *controller) DetachVolume(tenant string, volume string, attachment strin
 			continue
 		}
 
-		// send command to attach volume.
-		err = c.client.detachVolume(a.BlockID, a.InstanceID, i.NodeID)
+		i.StateLock.RLock()
```

Comment:
It would be nice to have a BAT test for this.  Does ciao-cli return an error if you try to detach a volume from a running instance?